content/blog/tyranny-of-nits/index.md

```diff
@@ -0,0 +1,208 @@
+Don't, by way of hypothetical example, pick a fight in on a stabilization report (stage 3), after years of implementation work, to argue about whether the feature should exist at all (a stage 1 concern).
+It doesn't matter if you think you're right (or even if you *are* right): this is fundamentally damaging to the project and to the team, and that will almost certainly outweigh the delta caused by a bad decision.
+
+## Setting the stakes: consequences and reversability
+
+The tricky part about this is knowing when to draw a close to the first two phases.
+When is an idea good enough to try out?
+When is a strategic set of compromises good enough to start implementing?
+
+When I think about this, I use two axes:
+
+1. **Impact:** how impactful is this? How many users will this impact? How central is this to what we're building? How visible is this change?
+2. **Reverseability:** how hard will it be to fix any mistakes that we make here? Can we just revert the change? Will there be things built on top of it?
+
+At Bevy, these are two of the most important factors involved in [marking a PR as Controversial](https://www.leafwing-studios.com/blog/triage-by-controversy/), forcing a higher standard of review.
+
+Obviously, these are important to consider for maintainers, when deciding whether to merge a PR or move a discussion forward.
+But they're really helpful for people in the thick of the process to consider to!
+Talk about these things clearly and explicitly when you're writing up your design docs or giving feedback.
+It'll help set the stakes, get you taken more seriously, and communicate to the poor author why you're being a pain in the ass.
+
+## Overruling objections: naive consensus is terrible actually
+
+Ultimately though, we want to move proposals through that lifecycle or kill them explicitly.
+
+Instead though, these concerns, both nits and serious architectural complaints, tend to just kill momentum and the thread goes dark.
+Why?
+
+It's because your project **can't make decisions.**
+You have to keep making decisions, day after day, or your momentum will die and everything will slowly rot.
+It doesn't even matter if they're "right": the very fact that a decision was made staves off this decay.
+
+Axiomatically: to make a decision, you have to both be able to say no, and to say yes.
+Open source projects are really good at saying no because they tend to run on naive consensus by default.
+The operating semantics of open source are simple:
+
+1. If everyone says "LGTM", the work gets merged.
+2. If someone reasis a concern, we wait for the author to fix it.
+3. If the author doesn't fix the concern, the PR stays open (or gets closed as stale).
+4. If the author comes back and says "that's not a problem actually", then we get orgnizational undefined behavior.
```

Comment:
Talking about these failure dynamics is interesting, but it should probably be moved earlier on. Adding a section about how the default behavior is this naieve consensus, and how that's bad, would help motivate and give context to the rest of the post.